### ISSUE TITLE
A error about should_prune()

### ISSUE THREAD

**Issue by blurLake** [optuna/optuna#1664]
## Optuna with openAI SB zoo. 
Hi, I am using Optuna and [openAI SB3 zoo]:https://stable-baselines3.readthedocs.io/en/master/guide/rl_zoo.html, to tune hyperparameters of one customized environment. By using command ```python3 train.py --algo td3 --env FullFilterEnv-v0 --gym-packages gym_environment -n 5 -optimize --n-trials 10 --n-jobs 2 --sampler random --pruner median```. Everything works fine in my macOS laptop. But the same code on a Ubuntu 18.04.1 LTS machine with the same command gives the following error: 

``` Trial 0 failed because of the following error: TypeError('should_prune() takes 1 positional argument but 2 were given',)
Traceback (most recent call last):
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/optuna/study.py", line 709, in _run_trial
    result = func(trial)
  File "/home/eejiiew/filter-zoo/rl-baselines3-zoo/utils/hyperparams_opt.py", line 95, in objective
    model.learn(n_timesteps, callback=eval_callback)
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/td3/td3.py", line 198, in learn
    reset_num_timesteps=reset_num_timesteps,
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 253, in learn
    log_interval=log_interval,
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 406, in collect_rollouts
    if callback.on_step() is False:
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/common/callbacks.py", line 92, in on_step
    return self._on_step()
  File "/home/eejiiew/filter-zoo/rl-baselines3-zoo/utils/callbacks.py", line 33, in _on_step
    if self.trial.should_prune(self.eval_idx):
TypeError: should_prune() takes 1 positional argument but 2 were given
Traceback (most recent call last):
  File "train.py", line 392, in <module>
    storage=args.storage, study_name=args.study_name, verbose=args.verbose)
  File "/home/eejiiew/filter-zoo/rl-baselines3-zoo/utils/hyperparams_opt.py", line 119, in hyperparam_optimization
    study.optimize(objective, n_trials=n_trials, n_jobs=n_jobs)
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/optuna/study.py", line 292, in optimize
    func, n_trials, timeout, catch, callbacks, gc_after_trial, None
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/optuna/study.py", line 654, in _optimize_sequential
    self._run_trial_and_callbacks(func, catch, callbacks, gc_after_trial)
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/optuna/study.py", line 685, in _run_trial_and_callbacks
    trial = self._run_trial(func, catch, gc_after_trial)
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/optuna/study.py", line 709, in _run_trial
    result = func(trial)
  File "/home/eejiiew/filter-zoo/rl-baselines3-zoo/utils/hyperparams_opt.py", line 95, in objective
    model.learn(n_timesteps, callback=eval_callback)
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/td3/td3.py", line 198, in learn
    reset_num_timesteps=reset_num_timesteps,
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 253, in learn
    log_interval=log_interval,
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 406, in collect_rollouts
    if callback.on_step() is False:
  File "/home/eejiiew/filter-zoo/venv/lib/python3.6/site-packages/stable_baselines3/common/callbacks.py", line 92, in on_step
    return self._on_step()
  File "/home/eejiiew/filter-zoo/rl-baselines3-zoo/utils/callbacks.py", line 33, in _on_step
    if self.trial.should_prune(self.eval_idx):
TypeError: should_prune() takes 1 positional argument but 2 were given
```

The train.py is given [here]: https://github.com/DLR-RM/rl-baselines3-zoo/blob/master/train.py

Could you give me some suggestions on this? Thank you very much.

**Comment by c-bata** [comment]
I guess you use older revision of rl-baselines3-zoo on an Ubuntu 18.04.1 LTS machine. The problem seems to be resolved 20 days ago.

https://github.com/DLR-RM/rl-baselines3-zoo/commit/3c0dec8abf91c1aac4547b1bc4be7ac8ebd052e4

Could you try to update the revision of rl-baselines3-zoo on an Ubuntu machine?

**Comment by blurLake** [comment]
Thanks for the reply. Indeed it is a version conflict. The error can be resolved either by downgrade optuna to 1.5.0, or do the modification in SB-zoo as in [here](https://github.com/DLR-RM/rl-baselines3-zoo/commit/3c0dec8abf91c1aac4547b1bc4be7ac8ebd052e4) about utils/callbacks.py.
